src/main/java/org/openstreetmap/atlas/checks/validation/intersections/AtGradeSignPostCheck.java

```diff
@@ -0,0 +1,644 @@
+     *         destination_sign relation
+     */
+    private FlaggedIntersection getIntersectionsWithIncompleteDestinationSignRelation(
+            final Map<AtlasEntity, Set<AtlasEntity>> roundAboutInEdgeToOutEdgeMap,
+            final Map<AtlasEntity, Set<AtlasEntity>> inEdgeToOutEdgeMap,
+            final Node intersectingNode, final Set<Relation> destinationSignRelations)
+    {
+        // If the node is part of destination sign relation, check if destination tag of the
+        // relation is missing or if there are any missing relations that the node could be part of
+        // based on from and to edges
+        final Set<AtlasEntity> entitiesToBeFlagged = new HashSet<>();
+        int instructionIndex = -1;
+        // Flag all roundabout edges that are missing destination sign relations or missing
+        // destination sign tags for existing relations
+        if (!roundAboutInEdgeToOutEdgeMap.isEmpty())
+        {
+            final Set<AtlasEntity> allRoundaboutEdgesMissingTagsOrRelations = this
+                    .getAllRoundaboutEdgesMissingTagsOrRelations(roundAboutInEdgeToOutEdgeMap,
+                            destinationSignRelations);
+            if (!allRoundaboutEdgesMissingTagsOrRelations.isEmpty())
+            {
+                entitiesToBeFlagged.addAll(allRoundaboutEdgesMissingTagsOrRelations);
+                instructionIndex = INSTRUCTION_INDEX_FOUR;
+            }
+        }
+        else
+        {
+            final Set<Relation> destinationSignRelationsMissingTag = this
+                    .getRelationsWithMissingDestinationTag(destinationSignRelations);
+            // Flag if destination sign tag is missing
+            if (!destinationSignRelationsMissingTag.isEmpty())
+            {
+                this.markAsFlagged(String.valueOf(intersectingNode.getIdentifier()));
+                instructionIndex = INSTRUCTION_INDEX_ONE;
+                entitiesToBeFlagged.addAll(destinationSignRelationsMissingTag);
+            }
+            else if (inEdgeToOutEdgeMap != null)
+            {
+                // If there are any missing destination sign relation that the node should be
+                // part of, flag it
+                final Set<AtlasEntity> connectedEdgesNotFormDestinationRelation = this
+                        .getConnectedEdgesNotFormDestinationRelation(inEdgeToOutEdgeMap,
+                                destinationSignRelations);
+                if (!connectedEdgesNotFormDestinationRelation.isEmpty())
+                {
+                    instructionIndex = INSTRUCTION_INDEX_TWO;
+                    entitiesToBeFlagged.addAll(connectedEdgesNotFormDestinationRelation);
+                }
+            }
+        }
+        return new FlaggedIntersection(instructionIndex, entitiesToBeFlagged);
+    }
+
+    /**
+     * Get items part of intersection that is not part of any destination_sign relations
```

Comment:
This description seems a little confusing,as it seems that the inputs are already filtered for items that aren't part of destination sign relations.

Reply:
This method returns intersections that do not have destination sign relations. The params for this method are not filtered prior based on destination_sign relation. I have slightly updated the comment. Let me know if it is still unclear. I will be happy to be more descriptive.

Reply:
Perhaps I am just missing something obvious here, but I am not seeing where the filtering for destination signs is happening. It looks the only filtering is for roundabout tags, and then everything gets added to the entitiesToBeFlagged set.

Reply:
My bad! This method is called only if there is no destination_sign relation. This method simply flags the items in the input hashmaps. Since the flag method was getting long, I just pulled out the section that flags the items in the maps to this method. I have renamed the method and replaced the java doc for the method.